src/pepr/operator/controllers/utils.ts

```diff
@@ -68,3 +69,36 @@ export async function purgeOrphans<T extends GenericClass>(
     }
   }
 }
+
+/**
+ * Purges Kubernetes resources of a specified kind within a namespace that match generation.
+ *
+ * @template T
+ * @param {string} generation - The generation label to retain.
+ * @param {string} namespace - The namespace to search for resources.
+ * @param {string} pkgName - The package name label to filter resources.
+ * @param {T} kind - The Kubernetes resource kind to purge.
+ * @param {Logger} log - Logger instance for logging debug messages.
+ * @returns {Promise<void>} - A promise that resolves when the operation is complete.
+ */
+export async function deleteChildren<T extends GenericClass>(
+  generation: string,
+  namespace: string,
+  pkgName: string,
+  kind: T,
+  log: Logger,
+) {
+  log.debug(`Deleting children of ${pkgName} in ${namespace} with generation ${generation}`);
+  const resources = await K8s(kind).InNamespace(namespace).WithLabel("uds/package", pkgName).Get();
+
+  for (const resource of resources.items) {
+    if (resource.metadata?.labels?.["uds/generation"] == generation) {
+      log.debug(resource, `Deleting child ${resource.kind!} ${resource.metadata!.name}`);
+      await K8s(kind).Delete(resource);
+    }
+  }
+}
```

Comment:
I think this is no longer used? I'm _more_ in favor of the current path so I think we can drop this delete func.